server/src/test/java/io/spine/server/delivery/DeliveryTest.java

```diff
@@ -215,12 +218,39 @@ public void returnOptionalEmptyIfPicked() {
         TenantAwareRunner.with(tenantId)
                          .run(() -> assertStatsMatch(delivery, index));
 
-        Optional<ShardProcessingSession> session =
-                registry.pickUp(index, ServerEnvironment.instance().nodeId());
-        assertThat(session).isPresent();
+        PickUpAck ack = registry.pickUp(index, ServerEnvironment.instance()
+                                                                .nodeId());
+        assertThat(ack.hasSession()).isTrue();
+        assertThat(monitor.failedToPickUp()).isEmpty();
+        assertThat(monitor.alreadyPicked()).containsExactly(index);
 
         TenantAwareRunner.with(tenantId)
                          .run(() -> assertStatsEmpty(delivery, index));
+        assertThat(monitor.failedToPickUp()).isEmpty();
+    }
+
+    @Test
+    @DisplayName("a single shard and notify monitor if shard pick-up failed for a technical reason")
+    public void notifyOnPickUpFailure() {
+        int shardCount = 1;
+        ShardedWorkRegistry registry = new ThrowingWorkRegistry();
+        FixedShardStrategy strategy = new FixedShardStrategy(shardCount);
+        MonitorUnderTest monitor = new MonitorUnderTest();
+        Delivery delivery = Delivery.newBuilder()
+                                    .setStrategy(strategy)
+                                    .setWorkRegistry(registry)
+                                    .setMonitor(monitor)
+                                    .build();
+        ServerEnvironment.when(Tests.class)
+                         .use(delivery);
```

Comment:
Please review the test code to make sure the delivery is reset after this test (no matter it failed or passed).

Reply:
The base class has the 
``` java
@AfterEach
public void tearDown() {
    ServerEnvironment.when(Tests.class)
                     .use(originalDelivery);
}
```
method that resets the delivery.